Automattic/wp-calypso
# Update/codemod client/signup

This change runs several codemods to achieve the following:

Before | After
------------ | -------------
`require( ... )` nested in blocks | All `require( ... )` moved to top block
`var ... = require( ... )` | `import ... from '...'`
`module.exports = ...` | `export default ...`
`this.translate` | `this.props.translate` via HOC
`React.createClass( ... )` | `class ThisComponentName extends React.Component`
`React.createClass( ... )` | `createReactClass( ... )` if unconvertible to `React.Component` subclass
`import { PropTypes } from 'react'` | `import PropTypes from 'prop-types'`

#### Testing instructions

1. Check out locally (`git checkout update/codemod-client/signup`) and start your server. You can also use a [live branch](https://calypso.live/?branch=update/codemod-client/signup)
2. Open the [Signup Page](http://calypso.localhost:3000/start)
3. Verify that the signup works as before!
